drivers/mge-hid.c

```diff
@@ -191,9 +192,28 @@ static char		mge_scratch_buf[20];
  * discharging.
  */
 
-#define			ABM_UNKNOWN  -1
-#define			ABM_DISABLED  0
-#define			ABM_ENABLED   1
+#define			ABM_UNKNOWN     -1
+#define			ABM_DISABLED     0
+#define			ABM_ENABLED      1
+
+/* Some untis like Eaton 9E Models that using "x.ChargerType=5" and "x.Status=2"
+ * instead of other units that has ABM by "x.ABMEnable and "x.Mode".
+ * NOTE: Eaton 5P unit has "x.ABMEnable=1 and "x.Mode=4","x.ChargerType=4" and "x.Status=19?" */
+#define			ABM_ENABLED_TYPE 4
+
+/* Define if we have battery.charger.type.status for 9E Models and others */
+#define			ABM_CHARGER_NO_STATUS -1
+#define			ABM_CHARGER_STATUS 1
+
+/* Define if we have battery.charger.mode.staus by "x.ABMEnable and "x.Mode" */
+#define			ABM_CHARGER_NO_MODE -1
+#define			ABM_CHARGER_MODE 1
+
+/* Internal flag to process battery.charger.type.status for 9E Models and others */
+static int advanced_battery_status = ABM_CHARGER_NO_STATUS;
+
+/* Internal flag to process battery.charger.mode.status by "x.ABMEnable and "x.Mode" */
+static int advanced_battery_mode = ABM_CHARGER_NO_MODE;
```

Comment:
I think this section complicates overall code readability a lot and the definitions are misleading into thinking these are device/HID return values (which may change in the future) when actually they are only internal states -1(unknown/uninitialized)/0(disabled)/1(enabled) that are better suited for recording with only an integer (in my opinion):

```
static int advanced_battery_status = -1;
static int advanced_battery_mode = -1;
```
Later we can check if the condition for `advanced_battery_status` is fulfilled and, if yes, set `advanced_battery_status = 1` and `advanced_battery_mode = 0` and vice-versa. Also, right now -1 is the same value for unknown and disabled, so better - I think - to treat -1 as unknown and 0 as disabled (as separate states).

Reply:
Maybe also rename these variables to be more descriptive, e.g. `abm_batt_status_present` or `abm_batt_mode_present`, because right now their names also kind of imply these are modes/statuses values returned from the device itself.